react/shippingStateMachine/shippingStateMachine.ts

```diff
@@ -49,7 +49,12 @@ const shippingStateMachine = Machine<
         },
       },
       selectAddress: {
-        initial: 'idle',
+        on: {
+          '': [
+            { target: '.idle', cond: 'hasAvailableAddresses' },
+            { target: 'createAddress' },
+          ],
+        },
```

Comment:
this doesn't seem to be working, I'm still getting the "too many recursion" error :-(

Reply:
fixed in 47790c6! I don't know exactly what the issue was, but I was able to get it working by moving this transition in the idle state itself and creating a guard for the inverse of `hasAvailableAddresses`